### ISSUE TITLE
Show the "Add image to linked Wikipedia article" icon for all images

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#3825]
**Summary:** 
PR #3783 adds a "Add image to linked Wikipedia article" icon to images uploaded via Nearby. As of now, the icon is only shown for the images that meet a certain criteria. I believe it's:
i. Image is uploaded via Nearby
ii. Has a linked Wiki article
iii. The linked wiki article doesn't have any images

For better UX, I think it would be nice if we show the wikipedia icon for all images. We can then enable/disable it based on the criteria. Whenever we disable it, we can show a proper reason for which it is disabled. This would help the users to easily get an idea about when we show and we don't show the icon.

Related issue: #872

**Comment by rhulkb27** [comment]
hi @sivaraam , what you propose is to remove these criteria and always show the "add image to linked wikipedia article" icon?

**Comment by sivaraam** [comment]
I'm suggesting to show the icon in all cases and decide to enable/disable it based on the criteria. When tapping on the icon when it is enabled one would get the dialog as usual. When tapping on the icon when it is disabled one should be shown the reason why it is disabled probably as a tooltip.

**Comment by rhulkb27** [comment]
@sivaraam I would really like to work on this issue, however, I am unable to repro the icon (I am not able to find an image that passes the criteria). Do you know if there is a way to force the app to give me the icon on a given image? 

Regardless, I can begin work on this now after I read through maskaravivek's code and find out how it works. The only issue (as stated earlier) is that I am unsure of how to test any code that I write without having an image that meets the criteria.

**Comment by sivaraam** [comment]
@rhulkb27 Thanks for your interest. For some background, I initially suggested this in PR #3783, but it wasn't addressed as people thought it needs to be discussed first and it didn't need to be part of the MVP. That's why I opened this issue. So, you can take this up if others agree to this change.

Regardless, some replies to your questions:

> Do you know if there is a way to force the app to give me the icon on a given image?

No, I don't think that's possible. Neither does it make sense to "force" the icon to be shown as that would mean we would not be satisfying the criteria for which the icons are shown which is not ideal for this task.

> The only issue (as stated earlier) is that I am unsure of how to test any code that I write without having an image that meets the criteria.

Yeah, it's tricky to get the icon triggered indeed. That's one of the reasons I was suggesting to show the icon for all images and disable it based on criteria. It would likely make the feature more discoverable. Anyways, try using the beta flavour of the app, find some nearby items which are associated with Wikipedia that lack images and upload an image to that nearby item via the nearby upload flow. Doing this the icon should be shown for the image you just uploaded.

**Comment by rhulkb27** [comment]
Alright, @sivaraam I'm confident that I will be able to implement this change, however as you said we need others to agree that this is a good change before I can begin working on it; do you know who exactly we need to give me the go-ahead from before I can start working on the issue? Maybe we can @ mention them here?

**Comment by sivaraam** [comment]
Let me ping some people: @misaochan @maskaravivek @neslihanturan

**Comment by rhulkb27** [comment]
@misaochan @maskaravivek @neslihanturan Hi, I just wanted to follow up on this issue; don't forget to let me know if you have any issues with this, otherwise, I will implement it tomorrow.

**Comment by misaochan** [comment]
What indicator shall we have for people to know at a quick glance whether or not it is enabled? Should it be greyed out if disabled? Otherwise it would defeat the purpose of having the icon in Contributions at all if users could not tell easily whether the option was available or not for a particular image without tapping it.

Aside from the others that you pinged, @nicolas-raoul 's input would be great. @rhulkb27 I would recommend waiting to implement until a consensus is reached on this.

**Comment by rhulkb27** [comment]
@misaochan I believe @sivaraam suggested to have a tooltip showing the reason why the image does not pass the criteria.

**Comment by sivaraam** [comment]
> Should it be greyed out if disabled?

I think it should be. Actually, I assumed it would be greyed out when I mentioned "disabled". Looks like I should have mentioned this explicitly.

> Otherwise it would defeat the purpose of having the icon in Contributions at all if users could not tell easily whether the option was available or not for a particular image without tapping it.

Yes, exactly. If we don't grey out then it would end up being very bad UX. I suggested the tooltip as an add-on to explain "_why_" the button is disabled and greyed-out.

**Comment by misaochan** [comment]
> I think it should be. Actually, I assumed it would be greyed out when I mentioned "disabled". Looks like I should have mentioned this explicitly.

Makes sense. I am okay with this change in that case. @neslihanturan what do you think?
